src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -205,7 +228,36 @@ internal List<T>.Enumerator GetEnumeratorFast()
 			=> _list.GetEnumerator();
 
 		private void RaiseVectorChanged(CollectionChange change, int index)
-			=> VectorChanged?.Invoke(this, new VectorChangedEventArgs(change, (uint)index));
+		{
```

Comment:
I'm wondering what will happen in this case:

```csharp
DependencyObjectCollection x = new();
x.VectorChanged += (_, _) =>
{
    var handle = new AutoResetEvent(false);

    _ = Task.Run(() =>
    {
        // Is this going to dead lock?
        x.VectorChanged += (_, _) => { };
        handle.Set();
    });

    handle.WaitOne();
};
```

Sorry for such crazy scenario. While it may be unlikely to see it in user apps, I think you may still be looking forward to having a correct implementation.

Reply:
That's fair, It's very unlikely, but possible nonetheless. I've adjusted it.

Reply:
> It's very unlikely

Yup. Hence mentioning "crazy scenario" 😄